internal/xds/translator/testdata/out/xds-ir/http-route-zonal-routing-weighted-clusters.endpoints.yaml

```diff
@@ -27,12 +27,18 @@
             address: 3.4.5.6
             portValue: 9090
       loadBalancingWeight: 1
+    loadBalancingWeight: 35
```

Comment:
this example should have created 1 cluster based on intent
https://github.com/envoyproxy/gateway/blob/main/internal/xds/translator/testdata/in/xds-ir/http-route-zonal-routing-weighted-clusters.yaml

but its creating 2 clusters and using a weighted cluster https://github.com/envoyproxy/gateway/blob/34a327b2c55ca72c10fb38285d4e200a6a4df6f9/internal/xds/translator/testdata/out/xds-ir/http-route-zonal-routing-weighted-clusters.routes.yaml#L15

https://github.com/envoyproxy/gateway/blob/34a327b2c55ca72c10fb38285d4e200a6a4df6f9/internal/ir/xds.go#L1520 needs to be updated, so we avoid this

Reply:
https://github.com/envoyproxy/gateway/blob/34a327b2c55ca72c10fb38285d4e200a6a4df6f9/internal/xds/translator/route.go#L251
and
https://github.com/envoyproxy/gateway/blob/34a327b2c55ca72c10fb38285d4e200a6a4df6f9/internal/xds/translator/translator.go#L535
is where its used